docs/design/automate-skus.md

```diff
@@ -0,0 +1,124 @@
+# Azure VM SKU and Locations Automation
+
+This document proposes a design to automate the updating of some data structures used to represent Azure constants in AKS Engine.
+
+## Problem
+
+AKS Engine represents some information about Azure in code constants or similar read-only data structures. Specifically, Azure data center location names and Virtual Machine SKU details are needed for validation and for enabling certain VM features.
+
+To stay in sync with Azure, these data structures must be refreshed periodically. To date, that has been a task for maintainers to perform on-demand. AKS Engine has often lagged behind Azure's introduction of new locations or VM SKUs.
+
+A python script will regenerate some, but not all of the data structures.
+
+## Goals
+
+- On a regular schedule, notify the AKS Engine team if new locations or SKUs exist
+- Automate the creation of the pull request with the new locations or SKUs
+
+## Non-goals
+
+## Proposal: scriptable aks-engine commands
+
+New developer-only commands in the `aks-engine` binary generate the necessary Go source code. The commands replace the existing python script and extend it to cover the accelerated networking capability list.
```

Comment:
Would it make sense to add another make target instead of and aks-engine commands?
If we do that it might be easier to periodically run the job in a devops pipeline or jenkins job to get a signal of when we need to perform updates.

Reply:
That signal could be that there were diffs between the checked in generated files and newly generated files.

Reply:
Yes, the presence of a diff in the code should be considered the signal that we need a PR. I forgot to put this explicitly in the doc, but :+1:.